README.md

```diff
@@ -21,128 +19,504 @@ composer require php-llm/llm-chain
+        int $number,
+    ): string {
+        // ...
+    }
+}
+```
+
+Please be aware, that this is only converted in a JSON Schema for the LLM to respect, but not validated by LLM Chain.
+
+### Response Streaming
+
+Since LLMs usually generate a response word by word, most of them also support streaming the response using Server Side
+Events. LLM Chain supports that by abstracting the conversion and returning a Generator as content of the response.
+
+```php
+use PhpLlm\LlmChain\Chain;
+use PhpLlm\LlmChain\Message\Message;
+use PhpLlm\LlmChain\Message\MessageBag;
+
+// Initialize Platform and LLM
+
+$chain = new Chain($llm);
+$messages = new MessageBag(
+    Message::forSystem('You are a thoughtful philosopher.'),
+    Message::ofUser('What is the purpose of an ant?'),
+);
+$response = $chain->call($messages, [
+    'stream' => true, // enable streaming of response text
+]);
+
+foreach ($response->getContent() as $word) {
+    echo $word;
+}
+```
+
+In a terminal application this generator can be used directly, but with a web app an additional layer like [Mercure](https://mercure.rocks)
+needs to be used.
+
+#### Code Examples
+
+1. **Streaming Claude**: [stream-claude-anthropic.php](examples/stream-claude-anthropic.php)
+1. **Streaming GPT**: [stream-gpt-openai.php](examples/stream-gpt-openai.php)
+
+### Image Processing
+
+Some LLMs also support images as input, which LLM Chain supports as `Content` type within the `UserMessage`:
+
+```php
+use PhpLlm\LlmChain\Message\Content\Image;
+use PhpLlm\LlmChain\Message\Message;
+use PhpLlm\LlmChain\Message\MessageBag;
+
+// Initialize Platoform, LLM & Chain
+
+$messages = new MessageBag(
+    Message::forSystem('You are an image analyzer bot that helps identify the content of images.'),
+    Message::ofUser(
+        'Describe the image as a comedian would do it.',
+        new Image(dirname(__DIR__).'/tests/Fixture/image.png'), // Path to an image file
+        new Image('https://foo.com/bar.png'), // URL to an image
+        new Image('data:image/png;base64,...'), // Data URL of an image
+    ),
+);
+$response = $chain->call($messages);
+```
+
+#### Code Examples
+
+1. **Image Description**: [image-describer-binary.php](examples/image-describer-binary.php) (with binary file)
+1. **Image Description**: [image-describer-url.php](examples/image-describer-url.php) (with URL)
+
+### Embeddings
+
+Creating embeddings of word, sentences or paragraphs is a typical use case around the interaction with LLMs and
+therefore LLM Chain implements a `EmbeddingsModel` interface with various models, see above.
+
+The standalone usage results in an `Vector` instance:
+
+```php
+use PhpLlm\LlmChain\OpenAI\Model\Embeddings;
+use PhpLlm\LlmChain\OpenAI\Model\Embeddings\Version;
+
+// Initialize Platform
+
+$embeddings = new Embeddings($platform, Version::textEmbedding3Small());
+
+$vector = $embeddings->create($textInput);
+
+dump($vector->getData()); // Array of float values
```

Comment:
```suggestion
var_dump($vector->getData()); // Array of float values
```

Reply:
We are in the lib not in the bundle so dump might not be available.

Reply:
The lib also has symfony/var-dumper as a dev dependency